vite.config.ts

```diff
@@ -7,21 +7,55 @@
 import react from "@vitejs/plugin-react";
 import fs from "node:fs";
 import path from "node:path";
-import { UserConfig, defineConfig, loadEnv } from "vite";
+import {
+  IndexHtmlTransformContext,
+  IndexHtmlTransformResult,
+  Plugin,
+  UserConfig,
+  defineConfig,
+  loadEnv,
+} from "vite";
 import { configDefaults } from "vitest/config";
 import svgr from "vite-plugin-svgr";
+import ejs from "ejs";
 
 // Support optionally pulling in external branding if the module is installed.
 const theme = "@microbit-foundation/ml-trainer-microbit";
 const external = `node_modules/${theme}`;
 const internal = "src/deployment/default";
 
+const viteEjsPlugin = (data: ejs.Data): Plugin => {
+  return {
+    name: "ejs",
+    transformIndexHtml: {
+      order: "pre",
+      handler: (
+        html: string,
+        _ctx: IndexHtmlTransformContext
+      ): IndexHtmlTransformResult => ejs.render(html, data),
+    },
+  };
+};
+
 export default defineConfig(({ mode }): UserConfig => {
   const commonEnv = loadEnv(mode, process.cwd(), "");
 
+  const themePath = fs.existsSync(external)
+    ? path.resolve(__dirname, external, "dist", "index.js")
+    : path.resolve(__dirname, internal, "index.tsx");
+  const file = fs.readFileSync(themePath, { encoding: "utf8" });
+  const match = /appNameFull: (?:"|')(.*)(?:"|'),*/.exec(file);
+  const appNameFull = match[1];
```

Comment:
Open to alternative, perhaps less fragile solutions. I'm not particularly fond of catering for both single and double quotes.

Reply:
Can we move the text we need here to a JSON file that index.js imports and we read directly?

Reply:
No, because ESM.

Reply:
Alternative approach on https://github.com/microbit-foundation/ml-trainer/pull/402